tests/python/relay/test_df_pattern.py

```diff
@@ -0,0 +1,783 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import tvm
+from tvm import relay
+from tvm.relay.df_pattern import *
+import numpy as np
+
+# NB: 1 corresponds to the C++ enum that specicfies this
+# we loose the type safety due to the Python/C++ calling
+# convention.
+K_ELEMWISE = 0
+K_BROADCAST = 1
+
+## NODE TESTS
+def test_expr_pattern():
+    ep = ExprPattern(relay.var('x', shape=(4, 1)))
+    print(ep)
+
+def test_var_pattern():
+    v = is_input("x")
+    print(v)
+
+def test_wildcard_pattern():
+    wc = wildcard()
+    print(wc)
+
+def test_CallPattern():
+    wc1 = wildcard()
+    wc2 = wildcard()
+    c = is_op("add")(wc1, wc2)
+    print(c)
+
+def test_TuplePattern():
+    wc1 = wildcard()
+    wc2 = wildcard()
+    t = TuplePattern([wc1, wc2])
+    print(t)
+
+def test_TupleGetItemPattern():
+    wc1 = wildcard()
+    wc2 = wildcard()
+    t = TuplePattern([wc1, wc2])
+    tgi = TupleGetItemPattern(t, 1)
+    print(tgi)
+
+def test_AltPattern():
+    is_add_or_sub = is_op('add') | is_op('subtract')
+    print(is_add_or_sub)
+
+def test_TypePattern():
+    ty_pat = has_type(relay.TensorType((10, 10), "float32"))
+    print(ty_pat)
```

Comment:
remove print, and use asserts instead, e.g. assert the type of the pattern, and the  subfields.